### PR TITLE
Logs Panel: JSON view in explore table

### LINKED ISSUE #79589: Logs Panel: Table UI - Format JSON in inspect view
One request that is coming from a customer is for a proper JSON format when displaying "Inspect value". If not in scope for the upcoming release, please consider for the next update. Thank you
image001
image002

_Originally posted by @eshward23 in https://github.com/grafana/grafana/issues/77469#issuecomment-1858179742_

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What is this feature?**
Exploration into setting JSON view in logs table when fields are JSON

image

Currently the logic is very simple, just checking each field value to see if it is JSON.

**Why do we need this feature?**
https://github.com/grafana/grafana/issues/79589

**Who is this feature for?**
Users of logs table

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/79589

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by svennergr** [comment]
@gtk-grafana Just came across this, and I think we talked about that, what's your opinion if the table itself can handle such things? It detects if the content is json and renders accordingly.

**Comment by gtk-grafana** [comment]
@svennergr I went ahead and moved it into the explore code

File: public/app/features/explore/Logs/LogsTableWrap.tsx [hunk 1]
```diff
@@ -55,8 +56,28 @@ export type FieldNameMeta = (InactiveFieldMeta | ActiveFieldMeta) & GenericMeta;
 type FieldName = string;
 type FieldNameMetaStore = Record<FieldName, FieldNameMeta>;
 
+function addJsonToDataFrame(dataFrame: DataFrame) {
+  const line = dataFrame.fields.find((field) => field.name === 'Line' || field.name === 'body');
+
+  if (line && line.values.length > 0 && line.values[0]) {
+    if (isLogLineJSON(line.values[0])) {
```
Comment (gtk-grafana) [inline public/app/features/explore/Logs/LogsTableWrap.tsx:63]: If the first log is valid JSON, but subsequent are not, they will be formatted as JSON, so when users inspect things might be broken.
Similarly, if the first log line is not valid JSON, but subsequent logs are, they will not be formatted as JSON.

**Comment by gtk-grafana** [comment]
I went ahead and started attempting to parse every field value, and only adding JSON config for fields that are completely JSON, takes <1ms to check everything, so maybe it's not too much work to check every field value after all!

**Comment by gtk-grafana** [comment]
Hmm, although the JSON field config doesn't get moved over when a table is added to a dashboard. 
image
image

**Comment by ryantxu** [comment]
Maybe https://grafana.com/docs/grafana/latest/panels-visualizations/query-transform-data/transform-data/#extract-fields

**Comment by gtk-grafana** [comment]
Closing in favor of https://github.com/grafana/grafana/pull/81152

**Pull request closed without merging**